### PR TITLE
Add TextArrayInput to edit arrays of strings

### PR DESCRIPTION
**Pull request by fzaninotto**
## Problem

Our various array input components don't support the simple case of editing an array of strings with no choices. 

## Solution

2 possibilities:

1. Tweak the `AutocompleteArrayInput` to support simple addition with `freeSolo`. This is not possible because this input heavily relies on suggestions, and can't take a value outside of its choices. Even the `onCreate` support uses a special choice. 
2. Implement a new, simpler  component based on MUI's `<Autocomplete>` component. 

I tried 1. but failed, so I went for 2.

## DX

```jsx
import { Create, SimpleForm, TextArrayInput, TextInput } from 'react-admin';

export const EmailCreate = () => (
    <Create>
        <SimpleForm>
            <TextArrayInput source="to" />
            <TextInput source="subject" />
            <TextInput source="body" multiline minRows={5} />
        </SimpleForm>
    </Create>
);
```

https://github.com/user-attachments/assets/2f8b28bb-7f5c-4bdc-908f-eecb10ebadfe

## How To Test

Launch the storybook and check out the `ra-ui-materialui/inputs/TextArrayInput stories` at http://localhost:9010/?path=/story/ra-ui-materialui-input-textarrayinput--basic. 

## To Do

- [x] Add component
- [x] Add stories
- [x] Add tests
- [x] Add Documentation
- [x] Reference it in other doc pages to make it discoverable

### CODE REVIEW

**Review by slax57 — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/TextArrayInput.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                    variant={variant}
+                    margin={margin}
+                    size={size}
+                />
+            )}
+            sx={sx}
+            {...field}
+            value={field.value || emptyArray} // Autocomplete does not accept null or undefined
+            onChange={(e, newValue: string[]) => field.onChange(newValue)}
+            {...props}
```
[the remark this answers is not in the document]
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/input/TextArrayInput.tsx:112]: TypeScript says no

**Comment by fzaninotto** [comment]
Review applied

File: examples/simple/src/data.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 export default {
     posts: [
         {
-            id: 1,
+            id: '1 1',
```
Comment (slax57) [inline examples/simple/src/data.tsx:4]: looks like you committed a change from a test unrelated to this PR

Comment (slax57) [reply examples/simple/src/data.tsx:4]:
```suggestion
            id: 1,
```

**Pull request merged**